src/util/system.cpp

```diff
@@ -398,8 +400,28 @@ void ArgsManager::WarnForSectionOnlyArgs()
         if (!found_result.first) continue;
 
         // otherwise, issue a warning
-        LogPrintf("Warning: Config setting for %s only applied on %s network when in [%s] section.\n", arg, m_network, m_network);
+        unsuitables.insert(arg);
     }
+    return unsuitables;
+}
+
+// Section names to be recognized in the config file.
+static const std::set<std::string> G_CONFIG_SECTIONS{
+    CBaseChainParams::REGTEST,
+    CBaseChainParams::TESTNET,
+    CBaseChainParams::MAIN
+};
```

Comment:
nit: Could reduce scope even further and move it into the function below? (Keeping the `static const` ...)

Reply:
Fixed.